### PR TITLE
[MIG] web_translate_dialog migrated to v10

### PR DESCRIPTION
**Pull request by JayVora-SerpentCS**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
It seems correct, Jay, thanks. I have just pushed latest commit made on 9.0.

**Review by pedrobaeza — APPROVED** [review]
Working flawlessly.

Thanks.

**Review by moylop260 — APPROVED** [review]
Could you check if new beta lints are valid https://travis-ci.org/OCA/web/jobs/214357692#L492?

File: web_translate_dialog/view/web_translate.xml [hunk 1]
```diff
@@ -1,18 +1,15 @@
...
+    <template id="assets_backend" name="web_translate_dialog assets"
+            inherit_id="web.assets_backend">
+        <xpath expr="." position="inside">
+            <script type="text/javascript"
+                    src="/web_translate_dialog/static/src/js/web_translate_dialog.js"/>
+            <link rel="stylesheet" id="translate-dialog-stylesheet"
+                href="/web_translate_dialog/static/src/css/base.css"/>
+        </xpath>
+    </template>
+</odoo>
```
Comment (moylop260) [inline web_translate_dialog/view/web_translate.xml]: Could you add a newline?

**Comment by pedrobaeza** [comment]
@Yajo please review JS changes

File: web_translate_dialog/static/src/js/web_translate_dialog.js [hunk 2]
```diff
@@ -5,19 +5,17 @@
...
-var FormView = require('web.FormView');
-var View = require('web.View');
+var data = require('web.data');
 var Dialog = require('web.Dialog');
-
-var _t = core._t;
+var FormView = require('web.FormView');
 var QWeb = core.qweb;
+var _t = core._t;
+var View = require('web.View');
```
Comment (yajo) [inline web_translate_dialog/static/src/js/web_translate_dialog.js]: Why this diff? You are doing the same, but it was alphabetically sorted before and now it is not... 🤔

File: web_translate_dialog/static/src/js/web_translate_dialog.js [hunk 3]
```diff
@@ -32,25 +30,27 @@ var translateDialog = Dialog.extend({
         this.view_type = parent.fields_view.type || '';
         this.$view_form = null;
         this.$sidebar_form = null;
-        if (!!field) {
+        var field_result = !!field;
+        if (field_result) {
```
Comment (yajo) [inline web_translate_dialog/static/src/js/web_translate_dialog.js]: Just put `if (field)`, please.

File: web_translate_dialog/static/src/js/web_translate_dialog.js [hunk 4]
```diff
@@ -61,13 +61,14 @@ var translateDialog = Dialog.extend({
             sup = this._super;
         // the template needs the languages
         $.when(this.languages_loaded).then(function() {
-            return sup.call(self);
+            return Reflect.apply(sup,self,[]);
+
```
Comment (yajo) [inline web_translate_dialog/static/src/js/web_translate_dialog.js]: Why not just `return $.when(this.languages_loaded).then($.proxy(this._super, this));`?

File: web_translate_dialog/static/src/js/web_translate_dialog.js [hunk 5]
```diff
@@ -118,14 +119,13 @@ var translateDialog = Dialog.extend({
     set_fields_values: function(lang, values) {
         var self = this;
         _.each(this.translatable_fields_keys, function(f) {
-            self.$el.find('.oe_translation_field[name="' + lang.code + '-' + f + '"]')
-                .val(values[f] || '')
-                .attr('data-value', values[f] || '');
+            self.$el.find('.oe_translation_field[name="' + lang.code + '-' + f + '"]').val(values[f] || '').attr('data-value', values[f] || '');
```
Comment (yajo) [inline web_translate_dialog/static/src/js/web_translate_dialog.js]: This is uglier than before.

File: web_translate_dialog/static/src/js/web_translate_dialog.js [hunk 6]
```diff
@@ -144,13 +144,7 @@ var translateDialog = Dialog.extend({
...
                 deff.resolve();
             } else {
-                self.view.dataset.call(
-                    'read',
-                    [[self.view.datarecord.id],
-                    self.translatable_fields_keys,
-                    self.view.dataset.get_context({
-                        'lang': lg.code
-                    })]).done(function (rows) {
+                self.view.dataset.call('read',[[self.view.datarecord.id],self.translatable_fields_keys,self.view.dataset.get_context({'lang': lg.code})]).done(function (rows) {
```
Comment (yajo) [inline web_translate_dialog/static/src/js/web_translate_dialog.js]: This is uglier than before.

File: web_translate_dialog/static/src/js/web_translate_dialog.js [hunk 7]
```diff
@@ -175,9 +169,7 @@ var translateDialog = Dialog.extend({
             }
             translation_mutex.exec(function() {
                 return new data.DataSet(self, self.view.dataset.model,
-                                        self.view.dataset.get_context())
-                    .write(self.view.datarecord.id, text,
-                           { context : { 'lang': code }});
+                                        self.view.dataset.get_context()).write(self.view.datarecord.id, text,{ context : { 'lang': code }});
```
Comment (yajo) [inline web_translate_dialog/static/src/js/web_translate_dialog.js]: This is uglier than before.

File: web_translate_dialog/static/src/js/web_translate_dialog.js [hunk 8]
```diff
@@ -225,5 +217,6 @@ common.AbstractField.include({
 return {
     translateDialog: translateDialog,
 };
+// odoo.define
+});
 
-}); // odoo.define
```
Comment (yajo) [inline web_translate_dialog/static/src/js/web_translate_dialog.js]: I hate these kind of comments. Seems absurd even to have to see a diff on them. I'd be glad if you just drop it.

File: web_translate_dialog/static/src/js/web_translate_dialog.js [hunk 9]
```diff
@@ -5,19 +5,17 @@
 odoo.define('web_translate_dialog.translate_dialog', function(require){
 "use strict";
 
-var _ = require('_');
-var $ = require('$');
+/*global Reflect*/
```
Comment (yajo) [inline web_translate_dialog/static/src/js/web_translate_dialog.js]: You can drop this comment if you fix something I comment you further below.

**Review by yajo — COMMENTED** [review]
I guess the changes come mostly from the new JS linter requirements, although they seem to make code worse than before IMHO.

I added some comments, but I'm not sure if best would be to:

- fix linter
- fix these comments
- leave JS as it was, since nothing *really* changed.

Thoughts? @moylop260

**Comment by JayVora-SerpentCS** [comment]
Will deal with these asap!

**Comment by JayVora-SerpentCS** [comment]
Are in a favour to go acc to https://travis-ci.org/OCA/web/jobs/218442515 ? Some JS lint checks if done acc to it, it is not always working on code.

**Review by yajo — APPROVED** [review]
You use weird JS indentation in some places, but I don't want to block this anymore just for that. At least not until we have a standard.

**Pull request merged**